src/Avalonia.Controls/AppBuilderBase.cs

```diff
@@ -21,33 +22,48 @@ public abstract class AppBuilderBase<TAppBuilder> where TAppBuilder : AppBuilder
         /// <summary>
         /// Gets or sets a method to call the initialize the runtime platform services (e. g. AssetLoader)
         /// </summary>
-        public Action RuntimePlatformServices { get; set; }
+        public Action RuntimePlatformServicesInitializer { get; private set; }
 
         /// <summary>
         /// Gets or sets the <see cref="Application"/> instance being initialized.
         /// </summary>
-        public Application Instance { get; set; }
+        public Application Instance { get; protected set; }
 
         /// <summary>
         /// Gets or sets a method to call the initialize the windowing subsystem.
         /// </summary>
-        public Action WindowingSubsystem { get; set; }
+        public Action WindowingSubsystemInitializer { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the name of the windowing subsystem to use.
+        /// </summary>
+        public string WindowingSubsystemName { get; set; }
```

Comment:
So there are now two ways of specifying a subsystem, both the initializer and the name? How do these interact? Would it be possible to just have the initializer and pass that a closure with the name?

If not, I think it should be documented how the two interact.